### PR TITLE
[Bug-fix][Recipes] Updating the run_name API to be run_name_prefix

### PR DESCRIPTION
**Pull request by sunishsheth2009**
## What changes are proposed in this pull request?
[Bug-fix][Recipes] Updating the run_name API to be run_name_prefix

## How is this patch tested?
- [x] Test added
- [x] Experiment Page:
image
 

## Does this PR change the documentation?
- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [x] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [x] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: tests/recipes/test_tracking_utils.py [hunk 1]
```diff
@@ -139,7 +142,11 @@ def test_get_recipe_tracking_config_returns_expected_config_on_databricks(
         )
         assert recipe_tracking_config.tracking_uri == (tracking_uri or default_tracking_uri)
         assert recipe_tracking_config.artifact_location == artifact_location
-        assert recipe_tracking_config.run_name == (run_name or None)
+        if run_name_prefix is not None:
+            assert run_name_prefix in recipe_tracking_config.run_name
+        else:
+            assert recipe_tracking_config.run_name is None
+        # assert recipe_tracking_config.run_name == (run_name_prefix or None)
```
Comment (harupy) [inline tests/recipes/test_tracking_utils.py:149]:
```suggestion
```

File: mlflow/recipes/utils/tracking.py [hunk 2]
```diff
@@ -30,6 +31,15 @@
 TrackingConfigType = TypeVar("TrackingConfig")
 
 
+def _get_run_name(run_name_prefix):
+    if run_name_prefix is None:
+        return None
+
+    sep = "-"
+    num = random.randint(0, 10**3)
```
Comment (harupy) [inline mlflow/recipes/utils/tracking.py:39]: I think this is going to conflict. For example, you run your recipe with the same run name prefix twice and this line generates the same integer. Can we use a short hash like `fa18aj7a`?

File: tests/recipes/test_tracking_utils.py [hunk 3]
```diff
@@ -77,7 +77,10 @@ def test_get_recipe_tracking_config_returns_expected_config(
     assert recipe_tracking_config.artifact_location == (
         artifact_location or default_artifact_location
     )
-    assert recipe_tracking_config.run_name == (run_name or None)
+    if run_name_prefix is not None:
+        assert run_name_prefix in recipe_tracking_config.run_name
+    else:
+        assert recipe_tracking_config.run_name is None
```
Comment (harupy) [inline tests/recipes/test_tracking_utils.py:83]:
```suggestion
    if run_name_prefix is None:
        assert recipe_tracking_config.run_name is None
    else:
        assert run_name_prefix in recipe_tracking_config.run_name
```

File: tests/recipes/test_tracking_utils.py [hunk 4]
```diff
@@ -139,7 +142,11 @@ def test_get_recipe_tracking_config_returns_expected_config_on_databricks(
         )
         assert recipe_tracking_config.tracking_uri == (tracking_uri or default_tracking_uri)
         assert recipe_tracking_config.artifact_location == artifact_location
-        assert recipe_tracking_config.run_name == (run_name or None)
+        if run_name_prefix is not None:
+            assert run_name_prefix in recipe_tracking_config.run_name
+        else:
+            assert recipe_tracking_config.run_name is None
```
Comment (harupy) [inline tests/recipes/test_tracking_utils.py:148]:
```suggestion
        if run_name_prefix is not None:
            assert recipe_tracking_config.run_name is None
        else:
            assert run_name_prefix in recipe_tracking_config.run_name
```

File: mlflow/recipes/utils/tracking.py (hunk 2 above)
Comment (BenWilson2) [reply mlflow/recipes/utils/tracking.py:39]: +1

**Review by harupy — APPROVED** [review]
LGTM once comments are addressed!

**Review by jinzhang21 — APPROVED** [review]

**Pull request merged**